Web/starter_project_iconics/components/common/blogCard.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { ReactElement } from 'react'
+import styles from '../../styles/blogCard.module.css'
+import { Blog } from '../../types/models'
+import { MdOutlineModeComment } from 'react-icons/md'
+
+import {
+  AiFillClockCircle,
+  AiFillCheckCircle,
+  AiFillWarning,
+} from 'react-icons/ai'
+
+interface Props {
+  card: Blog
+  feature: string
+}
+
+const BlogCard: React.FC<Props> = ({ card, feature }) => {
+  const handleReadMore = () => {
+    console.log('Read More button clicked')
+  }
+  const statusIcons: ReactElement[] = [
+    <AiFillCheckCircle key={1} size={20} className={styles.statIcon} />,
+    <AiFillWarning key={2} size={20} className={styles.statIcon} />,
+    <AiFillClockCircle key={3} size={20} className={styles.statIcon} />,
+  ]
+  const statusToNum: (stat: string) => number = (stat) => {
```

Comment:
[nit] This is a pretty good idea but it might be an even better approach to use a key-value pair instead of a list that you index to (you don't need to convert the strings "approved", "declined", [anything else] into numbers)

if for instance you had a object that had the strings as keys and the elements as values `{approved: <AiFillCheckCircle..., denied: <AiFillWarning..., none: <AiFillClockCircle...}`

then later on you can refer to it as `statusIcons[card.status]`
or you can even use `statusIcons[card.status] || statusIcons.none`, which means that it will always default to non if the status is not found as a key (meaning if it is not approved or denied)

yours works just fine, i am just suggesting because i feel it makes less code and it gets clearer which one each icon coresponds to when you define the list (usually using a list is not good in these cases because it might confuse a reader of the code as to why things are in that order. lists are for things you want to loop through if possible)

Reply:
I agree and it was pretty easy to change to dictionary and now it is changed